### DISCUSSION TITLE
Package utilization currently extremelly confusing when using Laravel Sanctum

### DISCUSSION THREAD

**Discussion by edsonfeimberg** [spatie/laravel-permission#2106] [General]
I have the impression that somethings should be a bit clearer for users using Laravel Sanctum. The origin of the biggest problems is this:

_"If your app uses only a single guard, but is not web (Laravel's default, which shows "first" in the auth config file) then change the order of your listed guards in your config/auth.php to list your primary guard as the default and as the first in the list of defined guards. While you're editing that file, best to remove any guards you don't use, too."_

Now, a Laravel Sanctum users should have something like this in config/auth:

```
   'defaults' => [
        'guard' => 'sanctum',
        'passwords' => 'users',
    ],

    'guards' => [
        'web' => [
            'driver' => 'session',
            'provider' => 'users',
        ],
    ],
```

The problem is that we can't make changes to this 'web' key in guards[] array, because Sanctum works on top of it. So problems arrive. For example, when trying to create a simple UserSeeder with a role of 'Super Admin' like this:

```
class UserSeeder extends Seeder
{
    /**
     * Run the database seeds.
     *
     * @return void
     */
    public function run()
    {
        $role = Role::create(['name' => 'Super Admin']);

        $user = User::create([
            'username' => 'Example',
            'email' => 'example@example.com',
            'password' => bcrypt('123456')
        ]);  

        $user->assignRole($role);
    }
}
```

We will get errors like:
```
Spatie\Permission\Exceptions\GuardDoesNotMatch
The given role or permission should use guard `web` instead of `sanctum`.
```

Am i missing something or the only current solution would be to always pass a `'guard_name' => 'web'` argument whenever creating new roles? Would manually setting `'guard_name' => 'web'` event grant the perfect function of the package since the default guard would be actually sanctum? Anyway, i have the impression that things could be a bit better explained to Sanctum users, since it is becoming kind of a default for Laravel Api development.

**Comment by erikn69** [comment]
I don't work with sanctum, maybe
```php
'defaults' => [
      'guard' => 'sanctum',
      'passwords' => 'users',
  ],

  'guards' => [
      'sanctum' => [
          'driver' => 'sanctum',
          'provider' => 'users',
      ],
      'web' => [
          'driver' => 'session',
          'provider' => 'users',
      ],
  ],
```

**Comment by angeljqv** [comment]
>Anyway, i have the impression that things could be a bit better explained to Sanctum users, since it is becoming kind of a default for Laravel Api development.

Feel free to make a PR for better documentation
